Drop-In/src/main/java/com/braintreepayments/api/SelectPaymentMethodParentFragment.java

```diff
@@ -65,37 +70,61 @@ public void transformPage(@NonNull View page, float position) {
             }
         });
 
+        requireActivity().getOnBackPressedDispatcher().addCallback(getViewLifecycleOwner(), new OnBackPressedCallback(true) {
+            @Override
+            public void handleOnBackPressed() {
+                int position = viewPager.getCurrentItem();
+                SelectPaymentMethodChildFragment visibleFragment =
+                    childFragmentList.getItem(position);
+
+                if (visibleFragment == VAULT_MANAGER) {
+                    onDismissVaultManager();
+                } else {
+                    sendDropInEvent(new DropInEvent(DropInEventType.CANCEL_DROPIN));
+                    remove();
+                }
+            }
+        });
+
+        Button backButton = view.findViewById(R.id.back_button);
+        backButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                sendDropInEvent(new DropInEvent(DropInEventType.CANCEL_DROPIN));
+            }
+        });
+
```

Comment:
Can we add some UI tests for this back button functionality / event dispatching?

Reply:
I was able to add a few back button tests. I can't figure out a way to test out the `popBackStack()` related tests right now though. I tried a good bit but it doesn't behave like expected.